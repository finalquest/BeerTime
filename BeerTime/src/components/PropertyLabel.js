import React from 'react';
import {
  View,
  StyleSheet,
} from 'react-native';
import PropTypes from 'prop-types';

import { StylePropType } from '../utils/PropTypes';
import Label from './Label';

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
  },
  name: {
    fontWeight: '400',
    textAlign: 'center',
    marginRight: 3,
  },
  value: {
    textAlign: 'center',
    flexShrink: 1,
  },
});

const PropertyLabel = ({
  style, name, value, separator, fontSize, nameStyles,
  valueStyles, Component, ellipsizeMode, numberOfLines,
}) => (
  <View style={[style, styles.container]}>
    <Label
      style={[nameStyles, { fontSize }]}
    >
      {`${name}${separator}`}
    </Label>
    <Component
      style={[valueStyles, { fontSize }]}
      numberOfLines={numberOfLines}
      ellipsizeMode={ellipsizeMode}
    >
      {value}
    </Component>
  </View>
);
PropertyLabel.propTypes = {
  style: StylePropType,
  name: PropTypes.string.isRequired,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  separator: PropTypes.string,
  fontSize: PropTypes.number,
  nameStyles: StylePropType,
  valueStyles: StylePropType,
  Component: PropTypes.func,
  ellipsizeMode: PropTypes.string,
  numberOfLines: PropTypes.number,
};
PropertyLabel.defaultProps = {
  style: {},
  separator: '',
  fontSize: 14,
  nameStyles: styles.name,
  valueStyles: styles.value,
  Component: Label,
  ellipsizeMode: 'tail',
  numberOfLines: 1,
};

export default PropertyLabel;
