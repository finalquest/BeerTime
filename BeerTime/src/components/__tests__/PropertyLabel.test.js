import React from 'react';
import { shallow } from 'enzyme';
import PropertyLabel from '../PropertyLabel';

const localize = { value: v => v };
describe('PropertyLabel test suite', () => {
  it('renders as expected', () => {
    const wrapper = shallow(<PropertyLabel name="test" value="v" />, localize);
    expect(wrapper).toMatchSnapshot();

    expect(wrapper.find('Label').at(0).prop('children')).toEqual('test');
  });
  it('supports separator', () => {
    const wrapper = shallow(<PropertyLabel name="test" value="v" separator=":" />, localize);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('Label').at(0).prop('children')).toEqual('test:');
  });
  it('supports fontSize', () => {
    const wrapper = shallow(<PropertyLabel name="test" value="v" fontSize={9} />, localize);
    expect(wrapper.find('Label').at(0).prop('style')).toContainEqual({ fontSize: 9 });
  });
  it('supports nameStyles', () => {
    const style = { };
    const wrapper = shallow(
      <PropertyLabel name="test" value="v" nameStyles={style} />,
      localize,
    );
    expect(wrapper.find('Label').at(0).prop('style')).toContainEqual(style);
  });
  it('supports valueStyles', () => {
    const style = { };
    const wrapper = shallow(
      <PropertyLabel name="test" value="v" valueStyles={style} />,
      localize,
    );
    expect(wrapper.find('Label').at(1).prop('style')).toContainEqual(style);
  });
  it('supports custom component for value render', () => {
    const MyComponent = ({ children }) => children;
    const wrapper = shallow(
      <PropertyLabel name="test" value="v" Component={MyComponent} />,
      localize,
    );
    expect(wrapper.find('MyComponent').prop('children')).toEqual('v');
  });
  it('supports numberOfLines', () => {
    const wrapper = shallow(
      <PropertyLabel name="test" value="v" numberOfLines={12} />,
      localize,
    );
    expect(wrapper.find('Label').at(1).prop('numberOfLines')).toEqual(12);
  });

  it('supports ellipsizeMode', () => {
    const wrapper = shallow(
      <PropertyLabel name="test" value="v" ellipsizeMode="head" />,
      localize,
    );
    expect(wrapper.find('Label').at(1).prop('ellipsizeMode')).toEqual('head');
  });
});
