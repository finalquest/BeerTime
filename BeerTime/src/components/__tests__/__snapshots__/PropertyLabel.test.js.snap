// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyLabel test suite renders as expected 1`] = `
<View
  style={
    Array [
      Object {},
      Object {
        "flexDirection": "row",
      },
    ]
  }
>
  <Label
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontWeight": "400",
          "marginRight": 3,
          "textAlign": "center",
        },
        Object {
          "fontSize": 14,
        },
      ]
    }
    uppercase={false}
  >
    test
  </Label>
  <Label
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      Array [
        Object {
          "flexShrink": 1,
          "textAlign": "center",
        },
        Object {
          "fontSize": 14,
        },
      ]
    }
    uppercase={false}
  >
    v
  </Label>
</View>
`;

exports[`PropertyLabel test suite supports separator 1`] = `
<View
  style={
    Array [
      Object {},
      Object {
        "flexDirection": "row",
      },
    ]
  }
>
  <Label
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontWeight": "400",
          "marginRight": 3,
          "textAlign": "center",
        },
        Object {
          "fontSize": 14,
        },
      ]
    }
    uppercase={false}
  >
    test:
  </Label>
  <Label
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      Array [
        Object {
          "flexShrink": 1,
          "textAlign": "center",
        },
        Object {
          "fontSize": 14,
        },
      ]
    }
    uppercase={false}
  >
    v
  </Label>
</View>
`;
